components/api/src/main/java/com/hotels/styx/api/support/UUIDGenerator.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (C) 2013-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.api.support;
+
+import java.util.UUID;
+
+import com.fasterxml.uuid.EthernetAddress;
+import com.fasterxml.uuid.Generators;
+import com.fasterxml.uuid.NoArgGenerator;
+import com.fasterxml.uuid.UUIDComparator;
+
+/**
+ * Utility class to make a {@link java.util.UUID}.
+ */
+public class UUIDGenerator {
+    private static final NoArgGenerator TIME_BASED_GENERATOR = Generators.timeBasedGenerator(EthernetAddress.fromInterface());
+
+    public static UUID makeUUID() {
+        return TIME_BASED_GENERATOR.generate();
+    }
+
+    public static boolean uuidsEqual(UUID u1, UUID u2) {
+        return UUIDComparator.staticCompare(u1, u2) == 0;
+    }
```

Comment:
From now on, do we need to start using (`uuidsEqual`) in in place of `Objects.equals`?

`Origin` and `BackendService` classes are still using the old `Object.equals` method in their `equalsTo` methods. Should these be replaced with `uuidsEqual`?

Reply:
The library author claims the UUID equals method is semi-vulnerable to long overflows. I'm not sure how relevant this is to normal Styx operations though. I just changed these 2 equals methods simply because they were in the same file. Are these(equals/hashcode) actually used for a purpose, or just good practice?

Reply:
I think the (unique) IDs should contribute to equality of HTTP messages only.

As discussed in other comments our usage of unique ids outside of HTTP messages is probably not justified.